lib/components/Box.tsx

```diff
@@ -47,203 +52,113 @@ type DangerDoNotUse = {
-  ]),
-  mb: mapUnitPropTo('marginBottom', halfUnit),
-  ml: mapUnitPropTo('marginLeft', halfUnit),
-  mr: mapUnitPropTo('marginRight', halfUnit),
-  mt: mapUnitPropTo('marginTop', halfUnit),
-  mx: mapDirectionalUnitPropTo('margin', halfUnit, ['Left', 'Right']),
-  my: mapDirectionalUnitPropTo('margin', halfUnit, ['Top', 'Bottom']),
-  // Padding
-  p: mapDirectionalUnitPropTo('padding', halfUnit, [
-    'Top',
-    'Bottom',
-    'Left',
-    'Right',
-  ]),
-  pb: mapUnitPropTo('paddingBottom', halfUnit),
-  pl: mapUnitPropTo('paddingLeft', halfUnit),
-  pr: mapUnitPropTo('paddingRight', halfUnit),
-  pt: mapUnitPropTo('paddingTop', halfUnit),
-  px: mapDirectionalUnitPropTo('padding', halfUnit, ['Left', 'Right']),
-  py: mapDirectionalUnitPropTo('padding', halfUnit, ['Top', 'Bottom']),
-  // Color props
-  color: mapColorPropTo('color'),
-  textColor: mapColorPropTo('color'),
-  backgroundColor: mapColorPropTo('backgroundColor'),
-} as const;
-
-// Boolean props
-const booleanStyleMap = {
-  bold: mapBooleanPropTo('fontWeight', 'bold'),
-  fillPositionedParent: (style, value) => {
-    if (value) {
-      style.position = 'absolute';
-      style.top = 0;
-      style.bottom = 0;
-      style.left = 0;
-      style.right = 0;
-    }
-  },
-  inline: mapBooleanPropTo('display', 'inline-block'),
-  italic: mapBooleanPropTo('fontStyle', 'italic'),
-  nowrap: mapBooleanPropTo('whiteSpace', 'nowrap'),
-  preserveWhitespace: mapBooleanPropTo('whiteSpace', 'pre-wrap'),
-} as const;
-
-export function computeBoxProps(props): Record<string, any> {
-  const computedProps: Record<string, any> = {};
-  const computedStyles: Record<string, string | number> = {};
-
-  // Compute props
-  for (const propName of Object.keys(props)) {
-    if (propName === 'style') {
-      continue;
-    }
-
-    const propValue = props[propName];
-
-    const mapPropToStyle =
-      stringStyleMap[propName] || booleanStyleMap[propName];
-
-    if (mapPropToStyle) {
-      mapPropToStyle(computedStyles, propValue);
-    } else {
-      computedProps[propName] = propValue;
-    }
-  }
-
-  // Merge computed styles and any directly provided styles
-  computedProps.style = { ...computedStyles, ...props.style };
-
-  return computedProps;
-}
-
-export function computeBoxClassName(props: BoxProps): string {
-  const color = props.textColor || props.color;
-  const backgroundColor = props.backgroundColor;
-  return classes([
-    isColorClass(color) && `color-${color}`,
-    isColorClass(backgroundColor) && `color-bg-${backgroundColor}`,
-  ]);
-}
-
 export function Box(props: BoxProps & DangerDoNotUse) {
   const { as = 'div', className, children, ...rest } = props;
 
-  // Compute class name and styles
   const computedClassName = className
     ? `${className} ${computeBoxClassName(rest)}`
     : computeBoxClassName(rest);
   const computedProps = computeBoxProps(rest);
 
-  // Render the component
   return createElement(
-    typeof as === 'string' ? as : 'div',
+    as,
```

Comment:
since this is defaulted to div, there's no real need to do a typeof check like before